templates/vue/src/components/modals/NodeModal/forms/ContentForm/MultipleChoiceForm.vue

```diff
@@ -0,0 +1,274 @@
+      this.question.answerTypes.multipleChoice.nextChoiceRowRadioId = newNextChoiceRowRadioId
+    },
+  },
+  created() {
+    if (
+      !this.question.answerTypes.multipleChoice.hasOwnProperty("checkboxArray") ||
+      !this.question.answerTypes.multipleChoice.hasOwnProperty("radioArray") ||
+      !this.question.answerTypes.multipleChoice.hasOwnProperty("useImages")
+    ) {
+      this.question.answerTypes.multipleChoice.checkboxArray = this.choiceRows
+      this.question.answerTypes.multipleChoice.radioArray = this.choiceRowsRadio
+      this.question.answerTypes.multipleChoice.nextChoiceRowId = this.nextChoiceRowId
+      this.question.answerTypes.multipleChoice.nextChoiceRowRadioId = this.nextChoiceRowRadioId
+      this.question.answerTypes.multipleChoice.useImages = this.useImages
+      this.question.answerTypes.multipleChoice.preSelectedCheckBoxOptions = this.preSelectedCheckBoxOptions
+      this.question.answerTypes.multipleChoice.preSelectedRadioOptions = this.preSelectedRadioOptions
+    } else {
+      this.choiceRows = this.question.answerTypes.multipleChoice.checkboxArray
+      this.choiceRowsRadio = this.question.answerTypes.multipleChoice.radioArray
+      this.nextChoiceRowId = this.question.answerTypes.multipleChoice.nextChoiceRowId
+      this.nextChoiceRowRadioId = this.question.answerTypes.multipleChoice.nextChoiceRowRadioId
+      this.useImages = this.question.answerTypes.multipleChoice.useImages
+      this.preSelectedCheckBoxOptions = this.question.answerTypes.multipleChoice.preSelectedCheckBoxOptions
+      this.preSelectedRadioOptions = this.question.answerTypes.multipleChoice.preSelectedRadioOptions
+    }
+  },
+  methods: {
+    addNewChoice: function() {
+      this.choiceRows.push({
+        id: this.nextChoiceRowId++,
+        imageurl: "",
+        value: "",
+      })
+    },
+    addNewChoiceRadio: function() {
+      this.choiceRowsRadio.push({
+        id: this.nextChoiceRowRadioId++,
+        imageurl: "",
+        value: "",
+      })
+    },
+    removeChoiceRowCheckbox: function(index, item) {
+      this.question.answerTypes.multipleChoice.checkboxArray.splice(index, 1)
+      for (let i = 0; i < this.preSelectedCheckBoxOptions.length; i++) {
+        if (item.id === this.preSelectedCheckBoxOptions[i]) {
+          this.preSelectedCheckBoxOptions.splice(i, 1)
+        }
+      }
+    },
+    removeChoiceRowRadio: function(index, item) {
+      this.question.answerTypes.multipleChoice.radioArray.splice(index, 1)
+      for (let i = 0; i < this.preSelectedRadioOptions.length; i++) {
+        if (item.id === this.preSelectedRadioOptions[i]) {
+          this.preSelectedRadioOptions.splice(i, 1)
+        }
+      }
+    },
+    updateOrderingCheckBoxArray(arr) {
+      this.question.answerTypes.multipleChoice.checkboxArray = arr
+    },
+    updateOrderingRadioArray(arr) {
+      this.question.answerTypes.multipleChoice.radioArray = arr
+    },
+  },
+}
+</script>
```

Comment:
I didn't write all the other cases, but this file will be simplified if we use the same data structure for both checkboxes and radios.

Reply:
Hi @wynnset, a lot of my functionality actually depends on managing two separate arrays for both the checkboxes and radios. For example, removing/adding an option, in the checkbox array will only remove/add the option in the checkbox array and not in the radio array. This is the same for adding images to a checkbox array, and radio array. In addition, having two lists is needed since we could pre-select more than one option in the checkbox array but could only pre-select one option in the radio array. There is a fair amount of functionality that is specific to a checkbox array and to a radio array.

Reply:
> Hi @wynnset, a lot of my functionality actually depends on managing two separate arrays for both the checkboxes and radios. For example, removing/adding an option, in the checkbox array will only remove/add the option in the checkbox array and not in the radio array. This is the same for adding images to a checkbox array, and radio array. 

Yes, I understand, but I'm suggesting we change that so it's just a single array so you wouldn't have to have all those duplications in your code. That will also allow authors to switch between radio and checkboxes if they want to. 
> In addition, having two lists is needed since we could pre-select more than one option in the checkbox array but could only pre-select one option in the radio array. 

@harry02031 you can easily figure this limit out by using the existing field you have set for the author to distinguish that they want to allow selection of multiple choices.